### PR TITLE
Qt: RenderWidget fixes

### PR DESCRIPTION
**Pull request by spycrab**
Fixes [issue #11015](https://bugs.dolphin-emu.org/issues/11015).

### CODE REVIEW

**Comment by JMC47** [comment]
Don't have much time right now, but, the two features added appear to work correctly on my laptop

File: Source/Core/DolphinQt2/RenderWidget.cpp [hunk 1]
```diff
@@ -70,6 +73,14 @@ void RenderWidget::OnHideCursorChanged()
   setCursor(Settings::Instance().GetHideCursor() ? Qt::BlankCursor : Qt::ArrowCursor);
 }
 
+void RenderWidget::OnKeepOnTopChanged(bool top)
+{
+  setWindowFlags(top ? windowFlags() | Qt::WindowStaysOnTopHint :
+                       windowFlags() ^ Qt::WindowStaysOnTopHint);
```
Comment (Tilka) [inline Source/Core/DolphinQt2/RenderWidget.cpp]: The second part should be `windowFlags() & ~Qt::WindowStaysOnTopHint` to not rely on the previous value.

**Comment by Tilka** [comment]
Merging this with @spycrab's promise to look into fixing the race conditions at some point.

File: Source/Core/DolphinQt2/RenderWidget.cpp [hunk 2]
```diff
@@ -116,9 +116,13 @@ bool RenderWidget::event(QEvent* event)
     break;
   case QEvent::WindowActivate:
     Host::GetInstance()->SetRenderFocus(true);
+    if (SConfig::GetInstance().m_PauseOnFocusLost && Core::GetState() == Core::State::Paused)
+      Core::SetState(Core::State::Running);
     break;
   case QEvent::WindowDeactivate:
     Host::GetInstance()->SetRenderFocus(false);
+    if (SConfig::GetInstance().m_PauseOnFocusLost && Core::GetState() == Core::State::Running)
+      Core::SetState(Core::State::Paused);
```
[the remark this answers is not in the document]
Comment (riking) [reply Source/Core/DolphinQt2/RenderWidget.cpp]: Yeah, at minimum it seems like this should be inside a RunOnCPUThread()?

Or even `bool Core::StateTransition(Core::State::Running, Core::State::Paused)` but that smells like something to include in an overhaul.

**Pull request merged**